### DISCUSSION TITLE
[Feature] View consolidated gallery

### DISCUSSION THREAD

**Discussion by ThisIsTheOnlyUsernameAvailable** [immich-app/immich#2846] [Feature Request]
### The feature

My wife and I will often attend the same event but take separate photos.  Whilst I can access her photos via partner sharing, it would be really nice to see both of our photos consolidated in a single view.

Is a consolidated view of *ALL* available photos be an easy feature to add?

### Platform

- [x] Server
- [X] Web
- [X] Mobile

**Comment by aero31aero** [comment]
This one might be easy to gloss over since Albums exist, but it has unique benefits in terms of less involved workflow from users; I can just know that all my family members have connected to wifi lately so next time I turn on the TV, I can immediately start viewing _all_ the photos they all took at the same event as I, without having to fiddle with all their phones and adding their pics to a shared album.

**Reply by schmitzkr** [reply]
I have a similar use case, my partner and I often take our kids on day trips, at the end of the trip we have both taken photos and we both want to share them with grandparents etc.  Currently (on android mobile), I need to share the photos from my timeline/gallery with those contacts and then click sharing, 'partners photo's' and then select one at a time (there isnt multi-select here) the photos/assets that they captured with those same contacts.  This also prevents me from easily making an album with those assets to share.
Ideally I would see any assets shared with me via partner sharing in my timeline/gallery so I can curate what I share with friends and family more easily.

**Comment by studylamp** [comment]
I'm working on migrating from another photo system and this is the biggest hurdle right now. The Explore and timeline need an option to include partner shared galleries.

**Comment by larsbloch** [comment]
I think this would be a really cool adition. You are able to see all partners images. It could be a toogle in the top of the photos page to chose whose images you want to show in timeline.

As others have stated i also have a usecase where me and my partner both use our phones for events and kids in general and it would be nice to see all pictures.

**Comment by ThisIsTheOnlyUsernameAvailable** [comment]
There is some overlap with another suggestion.  Linking here, in case either is considered for implementation.

https://github.com/immich-app/immich/discussions/3176

**Comment by harrismcc** [comment]
This would be a game changer for me, as current features like the map view and facial recognition make my family's large historical photo library much more usable and "explorable". However, this only works for whichever user is sort of the "host" of the big gallery, and everyone else in the family can't use those nice features.

Does anyone familiar with the codebase have an idea of how complicated something like this would be to implement? I'm a software engineer, and would be happy to contribute towards this.

**Reply by jrasm91** [reply]
I don't think it would be that complex to do on the web at least. The mobile app is not my area of expertise.

On the web, we use the two time bucket queries in the asset controller to populate the timeline. Right now it takes in an optional user id which means load my partners assets. We can probably change this to a list of user IDs. Then you could specify one or both and the database query could become In(ids). The main timeline would just need some sort of UI thing to let you select which partners to show/hide.

The map view is a separate endpoint, but we could add a similar implementation with userIds and a UI component to show/hide partners.

The server has an easy way to check for access to the requested users photos.

Feel free to make a focus-discussion on discord if you have any questions.

**Reply by jrasm91** [reply]
Sharing people between users for facial recognition is probably a bit more complex though. We'd have to figure out a way to "share" faces with other users so they can be matched when doing a similarity search.

**Comment by kaczkalolo** [comment]
Count me in. This is the only feature missing that keeps me using Google photos.

**Reply by jrasm91** [reply]
Variations of this comment seem to randomly come up on most topics at some point. Speaking as a core contributor, because it happens so regularly over such a wide-range of topics, it ends up being very meaningless to read/hear/see. Out of curiosity what is the purpose or intention of commenting with this message?

**Comment by alextran1502** [comment]
implemented

**Reply by ThisIsTheOnlyUsernameAvailable** [reply]
Thank you so much, @alextran1502 !  I really, really appreciate it.

**Reply by faryon93** [reply]
Just to clarify: A consolidated view in the gallery is only possible with the partner sharing option not with individually shared albums? Is that correct?

**Reply by masipcat** [reply]
Is it planned to be able to enable consolidated view for shared albums as well?
